endpoints/openrtb2/auction.go

```diff
@@ -1711,9 +1706,24 @@ func (deps *endpointDeps) processStoredRequests(ctx context.Context, requestJson
 	return resolvedRequest, impExtInfoMap, nil
 }
 
+func setImpExtInfoMap(imp json.RawMessage, impData ImpExtPrebidData, impExtInfoMap map[string]exchange.ImpExtInfo) error {
+	impId, err := jsonparser.GetString(imp, "id")
+	if err != nil {
+		if err.Error() == "Key path not found" {
+			return errors.New("request.imp[0] missing required field: \"id\"\n")
```

Comment:
We shouldn't be hardcoding `request.imp[0]` in the error message here. We should pass in the imp element number in the imp array to this function and add that to the error message